fml/platform/darwin/string_range_sanitization.h

```diff
@@ -0,0 +1,26 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_STRING_RANGE_SANITIZATION_H_
+#define FLUTTER_FML_STRING_RANGE_SANITIZATION_H_
+
+#include <Foundation/Foundation.h>
+
+namespace fml {
+
+// Helper to get the correct boundary of a character position in an NSString
+// given a byte index.
```

Comment:
Consider rephrasing as `Returns a range encompassing the grapheme cluster containing in which |index| is located.` 

Generally, starting with a third-person verb tends to encourage comments that focus on what the code does. I'd also avoid the use of the word 'character' as it's pretty ambiguous. Similar below.

Reply:
The underlying impelementation throws an exception (`NSInvalidArgumentException`).  Is that fair for us to just do?